### DISCUSSION TITLE
”SparkFun Thing Plus Matter” MGM240P/EFR32MG24 based development board sold for $25

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#1172] [General]
SparkFun has released an MGM240P/EFR32MG24-based development board called ”SparkFun Thing Plus Matter” selling for $25 + shipping:

https://www.sparkfun.com/products/20270

https://youtu.be/TO45V9c-yFw

https://learn.sparkfun.com/tutorials/sparkfun-thing-plus-matter---mgm240p-hookup-guide/

Front: ![image](https://user-images.githubusercontent.com/6320001/223357223-50678254-a97a-4fbf-b6b4-714e00c30e92.png)      Back: ![image](https://user-images.githubusercontent.com/6320001/223357267-f526b65f-d06b-4727-962e-d5530efd038d.png)

### SparkFun Thing Plus Matter (MGM240P/EFR32MG24) specifications:

https://www.cnx-software.com/2023/03/03/sparkfun-thing-plus-matter-mgm240p-board-supports-matter-zigbee-openthread-and-bluetooth-low-energy-5-3/

* [MGM240P wireless module](https://www.silabs.com/wireless/zigbee/efr32mg24-series-2-modules/device.mgm240pa32vnn?tab=specs) from Silicon Labs
  * SoC –  SiLabs [EFR32MG24](https://www.cnx-software.com/2022/01/24/silicon-labs-bg24-and-mg24-2-4-ghz-wireless-mcus-quadruple-ai-performance-at-a-fraction-of-the-energy/) microcontroller
    * CPU – 32-bit Arm Cortex-M33 @ 39 MHz (MCU/module specs say up to 78 MHz, but maybe it’s clocked lower in this board for some reason, maybe lower power consumption?)
    * Memory/Storage – 1536kB Flash Memory, 256kB RAM
    * Wireless – 802.15.4 wireless protocols ([Zigbee](https://www.cnx-software.com/news/zigbee) and OpenThread) and Bluetooth Low     *     * Energy 5.3; Matter-ready
    * Secure Vault Support
    * Built-in Antenna
* Storage – **MicroSD card socket**
* USB – 1x USB Type-C port for programming, debugging, power, and battery charging
* Expansion
  * Two rows of header with 21x GPIO configurable as I2C, SPI, UART, I2S, 6x analog inputs, plus Reset, Enable, and power signals
  * 4-pin JST Qwiic connector
* Debugging
  * EFM32GG12B410F1024GL120-A microcontroller used as J-Link programmer and debugging IC
  * Unpopulated Mini Simplicity connector to connect an external debugger
* Misc
  * Reset button
  * 4x LEDs (Power, Status, Charging, and Debugging)
  * Solder jumpers for customization
* Power Supply
  * 5V via USB-C port
  * 2-Pin JST connector for a single LiPo battery (not included) with MC73831 LiPo charger and MAX17048 LiPo fuel gauge
  * Power consumption  – 15µA when MGM240P is in Low Power mode
* Dimensions – 5.84 x 22.9 cm compatible with the Thing Plus form factor (Feather-compatible); 2x mounting holes

**Comment by MattWestb** [comment]
For the moment is not so many Silabs chip / modules that can running native Matter over Thread then they is not having enough of RAM for high level applications so only the MG12/MG13 and MG24 then MG22 is normally having to less RAM and the MG21 was taking out in the GSDK 4.2.X. Matter support of the same reason (and MG23 was never coming out on the market of some not known reason).

And the great Thunderboard Sense 2 is out of production but was having great demo hardware and was at good price.

**Comment by Hedda** [comment]
FYI, spotted at [cnx-software.com tech blog](https://www.cnx-software.com/2024/03/26/arduino-nano-matter-board-specifications-and-price/) and looks like another nice MGM240 hardware board for prototypes and testing, etc.

"**_Arduino Nano Matter_**" development board with USB is based on the MGM240S SiP module (containing Silicon Labs EFR32MG24 SoC) 

Similar to MGM240P based [Sparkfun Things Plus Matter](https://www.sparkfun.com/products/20270) board so guess requires a Gecko bootloader to be flashed via SWD first if want to be able to flash via USB later? See example -> https://github.com/darkxst/silabs-firmware-builder/pull/31

Anyway, **Arduino Nano Matter**"  (SKU: ABX00112) is now available for $25(US).

* https://store-usa.arduino.cc/pages/nano-matter
  * https://docs.arduino.cc/hardware/nano-matter/#features
    * https://docs.arduino.cc/resources/pinouts/ABX00112-full-pinout.pdf
    * https://docs.arduino.cc/resources/datasheets/ABX00112-datasheet.pdf
    * https://docs.arduino.cc/resources/schematics/ABX00112-schematics.pdf
      * https://docs.arduino.cc/hardware/nano-matter/
      * https://docs.arduino.cc/tutorials/nano-matter/user-manual/

Key Specifications:
 
* Silabs MGM240S module MGM240SD22VNA (MGM240SD22VNA2R U5)
  * MCU Core = ARM Cortex-M33		
    * Core Frequency (MHz) = 78		
  * Flash (kB) = 1536
  * RAM (kB) = 256
  * IEEE 802.15.4 Radio Transmit Output Power (dBm) = 10
  * Silicon Labs Secure Vault™ = High
* Antenna = 2.4GHz Ceramic Chip Antenna surface mounted component on-board
* USB – USB-C port for power and data
  * USB bridge via Microchip ATSAM-D11-D14A-MUT (a.k.a. Atmel SAMD11 / Atmel SAM D11) chip on-board
* Debugging – JTAG/SWD debug port (accessible through the board’s test pads)
* Peripherals
  * Analog – 12-bit ADC (x19), up to 12-bit DAC (x2)
  *Digital – 22x GPIO pins (2x UART, 2x I2C, 2x SPI, 5x PWM)
  *External interrupts are available for all digital pins
* Misc
  * 2x push buttons (RST and user button)
  * RGB LED
  * Castellated pins for integrating the board into final solutions
* Power
  ' 5V via USB-C or header connector pins (VIN, IN5V)
  ' Operating Voltage: 3.3V
* Dimensions – 45 x 18 mm

MGM240S module - MGM240SD22VNA2R (U5):
 * https://www.silabs.com/wireless/zigbee/efr32mg24-series-2-modules/device.mgm240sd22vna?tab=specs
* https://www.silabs.com/documents/public/data-sheets/mgm240s-datasheet.pdf

EFR32MG24 SoC:
* https://www.silabs.com/wireless/zigbee/efr32mg24-series-2-socs
* https://www.silabs.com/documents/public/data-sheets/efr32mg24-datasheet.pdf

![image](https://github.com/darkxst/silabs-firmware-builder/assets/6320001/6080f8f3-676d-441d-af09-50b84e7fa08a)

![image](https://github.com/darkxst/silabs-firmware-builder/assets/6320001/b66771df-5878-4d15-a334-1d91fa7661cb)

![image](https://github.com/darkxst/silabs-firmware-builder/assets/6320001/622f7868-aa96-418d-b868-48b17691508f)

![image](https://github.com/darkxst/silabs-firmware-builder/assets/6320001/ea37a1c3-4523-4e81-bbf2-92d39af6a61c)

![image](https://github.com/darkxst/silabs-firmware-builder/assets/6320001/059f44d8-762d-46d0-ab85-6c8d71562fd9)
